tests/Doctrine/Tests/Common/Cache/CacheTest.php

```diff
@@ -236,6 +233,30 @@ public function provideCacheIds()
         );
     }
 
+    public function testLifetime()
+    {
+        $cache = $this->_getCacheDriver();
+        $cache->save('expire', 'value', 1);
+        $this->assertTrue($cache->contains('expire'), 'Data should not be expired yet');
+        sleep(2);
```

Comment:
As you can see the test already reveaed two inconsistencies in the cache implementations.